### ISSUE TITLE
Test failure Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactoryTest.CallSitesAreUniquePerServiceTypeAndSlot

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#52880]
Run: [runtime 20210517.88](https://dev.azure.com/dnceng/public/_build/results?buildId=1142790&view=ms.vss-test-web.build-test-results-tab&runId=34631102&resultId=100882&paneView=debug)

Failed test:
```
net6.0-Browser-Release-wasm-Mono_Release-wasmtestonbrowser-Ubuntu.1804.Amd64.Open

- Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactoryTest.CallSitesAreUniquePerServiceTypeAndSlot

```
**Error message:**
```
System.PlatformNotSupportedException : Cannot wait on monitors on this runtime.


Stack trace
   at System.Threading.Monitor.ObjWait(Int32 millisecondsTimeout, Object obj)
   at System.Threading.Monitor.Wait(Object obj, Int32 millisecondsTimeout)
   at System.Threading.ManualResetEventSlim.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.WaitAllBlockingCore(List`1 tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.WaitAllCore(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.WaitAll(Task[] tasks)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactoryTest.CallSitesAreUniquePerServiceTypeAndSlot()
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

**Comment by davidfowl** [comment]
@maryamariyan I think we need to skip this test on browsers?

**Comment by davidfowl** [comment]
@elinor-fung this is fixed with your PR right?

**Comment by elinor-fung** [comment]
Yes - should be fixed with #52881
